src/dbnode/storage/shard.go

```diff
@@ -1499,6 +1503,25 @@ func (s *dbShard) insertSeriesBatch(inserts []dbShardInsert) error {
 
 		if inserts[i].opts.hasPendingIndexing {
 			pendingIndex := inserts[i].opts.pendingIndex
+
+			// Don't insert cold writes into the index insert queue.
+			if s.coldWritesEnabled && writeType == series.ColdWrite {
+				indexBlockStart := s.reverseIndex.BlockStartForWriteTime(pendingIndex.timestamp)
+				// NB(bodu): We finalize this entry w/o decrementing the entry ref count
+				// unless the entry is marked as having its ref count incremented.
+				// Entries in the shard insert queue are either of:
+				// - new entries
+				// - existing entries that we've taken a ref on (marked as entryRefCountIncremented)
+				// Also, currently `releaseEntryRef` will always be true if `hasPendingIndex` is also true
+				// but we can do the check below just to be safe in case things change.
+				entryFinalizeFn := entry.OnIndexFinalizeNoRef
+				if releaseEntryRef {
+					entryFinalizeFn = entry.OnIndexFinalize
+				}
+				entryFinalizeFn(indexBlockStart)
+				continue
```

Comment:
Hm, instead of continuing can we just not execute the other code? Main concern is things like `hasPendingRetrievedBlock` potentially needing to be executed below.